GridKit/AutomaticDifferentiation/DependencyTracking/Variable.hpp

```diff
@@ -254,6 +254,12 @@ namespace GridKit
       inline Variable& operator/=(const double& rhs);
       inline Variable& operator/=(const Variable& rhs);
 
+      // conversion operator
+      inline operator double() const
+      {
+        return value_;
+      }
+
```

Comment:
```suggestion
      // conversion operator
      explicit inline operator double() const
      {
        return value_;
      }
```
Since you're using `static_cast` elsewhere, maybe it's a good idea to mark this operator as `explicit`, that way we can avoid a bug that looks something like this:

```cpp
real_type x = y[0];
...
f[1] = x * y[1];
```

Reply:
Addressed in [427d82c](https://github.com/ORNL/GridKit/pull/305/commits/427d82cc5cc499e65d170c3c251cd3e1244bf08e)